nservicebus/messaging/message-type-detection.md

```diff
@@ -1,24 +1,24 @@
 ---
 title: Message type detection
 summary: Describes the rules for mapping a transport message to a message type
-reviewed: 2021-06-05
+reviewed: 2023-08-04
 component: Core
+related:
+- samples/consumer-driven-contracts
 ---
 
-In order to invoke the correct message handlers for incoming messages, NServiceBus must be able to map the incoming transport message to a [message type](/nservicebus/messaging/messages-events-commands.md).
+NServiceBus sets the [`NServiceBus.EnclosedMessageTypes` header](/nservicebus/messaging/headers.md#serialization-headers-nservicebus-enclosedmessagetypes) to the [AssemblyQualifiedName](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname) of the emitted message type when sending, publishing or replying.
+
+It is possible to change the enclosed message type header for all emitted messages by using a behavior. The [consumder-driven-contracts sample](/samples/consumer-driven-contracts) shows how to manipulate the header for outgoing messages.
```

Comment:
Should this provide some information under what circumstances this would be a reasonable behavior to add?

```suggestion
It is possible to change the enclosed message type header for all emitted messages by using a behavior. The [consumer-driven-contracts sample](/samples/consumer-driven-contracts) shows how to manipulate the header for outgoing messages.
```

Reply:
Good question. I thought that is implied by linking to the sample and curious people can read that context there but happy to expand